Add new tail command options (#2095)
cloudflare/cloudflare-docs: products/workers/src/content/cli-wrangler/commands.md

Starts a log tailing session for a deployed Worker.

```sh
$ wrangler tail [--format $FORMAT] [--status $STATUS] [OPTIONS]
```

<Definitions>

- `--format $FORMAT` <Type>json|pretty</Type>
  - The format of the log entries.
- `--header $HEADER`
  - Filter by HTTP header
- `--ip-address $IP`
  - Filter by IP address ("self" to filter your own IP address)
- `--method $METHOD`
  - Filter by HTTP method
- `--sampling-rate $RATE` 
  - Adds a sampling rate (0.01 for 1%) [default: 1]
- `--search $SEARCH`

>   - Filter by a text match in console.log messages

Reviewer: ```suggestion
  - Filter by a text match in `console.log` messages
```

- `--status $STATUS`

>   - Filter by invocation status [possible values: ok, error, canceled]

Reviewer 2: Is this exactly how the values show up in the terminal or are they capitalized? (like 'OK, ERROR, CANCELED')

Author: They are lowercased.

>   - Filter by invocation status [possible values: ok, error, canceled]

Reviewer: invocation status is different than http status, right? how would a user run into error/canceled states?

Author: This one is referring to invocation status.

</Definitions>

After starting `wrangler tail` in a directory with a project, you will receive a live feed of console and exception logs for each request your Worker receives.

> Like all Wrangler commands, run `wrangler tail` from your Worker’s root directory (i.e. the directory with your `wrangler.toml`).

Reviewer 2: ```suggestion
Like all Wrangler commands, run `wrangler tail` from your Worker’s root directory (i.e., the directory with your `wrangler.toml`).
```

  - Filter by invocation status [possible values: ok, error, canceled]

</Definitions>

After starting `wrangler tail` in a directory with a project, you will receive a live feed of console and exception logs for each request your Worker receives.

Like all Wrangler commands, run `wrangler tail` from your Worker’s root directory (i.e. the directory with your `wrangler.toml`).

### Dependencies

Wrangler tail uses cloudflared under the hood. If you are already using cloudflared, be sure you have installed the latest version. Otherwise, follow the [getting started guide](https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/install-and-setup) for Cloudflare Tunnel.
`wrangler tail` will register a tailing session for your Worker, and start a server on `localhost` with a [tunnel](https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/install-and-setup) that listens for incoming log requests from your Worker.

> <Aside type="warning" header="Legacy issues with existing cloudflared configuration">

Reviewer: nvm, i see your comment on this already

For versions before Wrangler 1.19.0, `wrangler tail` will not work with existing `cloudflared` configuration on a local machine. This is a well-known issue, [tracked in this Github issue](https://github.com/cloudflare/wrangler/issues/1844). To update your Wrangler version, refer to the [update documentation](/cli-wrangler/install-update#update).

To apply a temporary fix, rename your `cloudflared` config to allow `wrangler tail` to work correctly. 

```sh
# Move config file when using `wrangler tail`. 
# This will temporarily disable `cloudflared`. 
$ mv ~/.cloudflared/config.yml ~/.cloudflared/config.yml.disabled

# Move file back when you need to use `cloudflared`.
$ mv ~/.cloudflared/config.yml.disabled ~/.cloudflared/config.yml
```

</Aside>

> 

Reviewer 2: Kristian and I decided to leave this in temporarily in case people had not updated their wrangler version yet. I'm not sure how user experience goes, but is updating wrangler a pretty common-sense action for people to take or do they hold off on it? Or is the new update of not needing cloudflared make this whole section pointless? If it's pointless, I'm okay w deleting the whole thing.

Reviewer 2: Also I know it went to *Changes approved* state on PR but need Kristian or Luke to look this PR over before merge.

Reviewer: it's been about two weeks since we updated this page to mention the new version of wrangler. maybe instead of removing this entirely, we can update it to link to the upcoming blog post, and say "We've recently shipped an overhauled version of wrangler tail that removes a dependency on cloudflared. Make sure to update (link) and check out the blog post (link) to learn more." rough stab at copy, but the idea will be instead of giving them a list of instructions, we can start to point them towards the new features as incentive to upgrade

Author: Gotcha. What should we do before the blog post is out?

Author: ![image](https://user-images.githubusercontent.com/19178821/129974687-2fc16b08-4e75-4520-9e02-f9c764994470.png)

We need to deprecate the old version. I think that letting folks know that the <1.19.0 depends on cloudflared is good enough if they want to debug an earlier version

Starts a log tailing session for a deployed Worker.

```sh
$ wrangler tail [--format $FORMAT] [--status $STATUS] [OPTIONS]
```

<Definitions>

- `--format $FORMAT` <Type>json|pretty</Type>
  - The format of the log entries.
- `--status $STATUS`
  - Filter by invocation status [possible values: ok, error, canceled]
- `--header $HEADER`
  - Filter by HTTP header
- `--method $METHOD`
  - Filter by HTTP method
- `--sampling-rate $RATE` 
  - Adds a percentage of requests to log sampling rate
- `--search $SEARCH`
  - Filter by a text match in `console.log` messages

</Definitions>

After starting `wrangler tail` in a directory with a project, you will receive a live feed of console and exception logs for each request your Worker receives.

Like all Wrangler commands, run `wrangler tail` from your Worker’s root directory (i.e., the directory with your `wrangler.toml`).

<Aside type="warning" header="Legacy issues with existing cloudflared configuration">

> wrangler tail versions <1.19.0 use cloudflared under the hood. We recommend [updating to the latest wrangler version](/cli-wrangler/install-update#update).

Reviewer 2: ```suggestion
wrangler tail versions <1.19.0 use cloudflared under the hood. Cloudflare recommends [updating to the latest wrangler version](/cli-wrangler/install-update#update).
```
